src/runez/system.py

```diff
@@ -387,27 +388,27 @@ def get_version(mod, default="0.0.0", fatal=False, logger=False):
     if name and isinstance(name, str):
         top_level = name.partition(".")[0] if isinstance(name, str) else name
         last_exception = None
+        metadata_version = _metadata_version_function()
+        if metadata_version:
+            try:
+                v = metadata_version(top_level)
+                if v:
+                    return v
 
-        try:
-            from importlib import metadata  # requires py3.8+
-
-            version = metadata.version(top_level)
-            if version:
-                return version
-
-        except (ImportError, Exception) as e:
-            last_exception = e
+            except (ImportError, Exception) as e:
```

Comment:
ImportError inherits from Exception, why do you list it separately?

Reply:
Ah indeed, fixed. Originally this was mixing both import errors (py version too low for `importlib.metadata.version` for example), and exceptions that may happen within the call itself.